From 1a5b810c2093e2e9043615cdd7fee6409e422519 Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Thu, 26 Sep 2024 16:26:30 +0800
Subject: [PATCH 219/265] mtk: mt76: mt7996: fix the timing to send sta_rec_bf

Send sta_rec_bfer and sta_rec_bfee tlv only when the connection state
changes to CONN_STATE_CONNECT.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>

---
 mt7996/mcu.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index 1f903acb..090e4887 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -3256,8 +3256,12 @@ int mt7996_mcu_add_sta(struct mt7996_dev *dev, struct ieee80211_bss_conf *conf,
 
 		/* starec hdrt mode */
 		mt7996_mcu_sta_hdrt_tlv(dev, skb);
-		/* starec bfer */
-		mt7996_mcu_sta_bfer_tlv(dev, skb, conf, mconf, link_sta);
+		if (conn_state == CONN_STATE_CONNECT) {
+			/* starec bfer */
+			mt7996_mcu_sta_bfer_tlv(dev, skb, conf, mconf, link_sta);
+			/* starec bfee */
+			mt7996_mcu_sta_bfee_tlv(dev, skb, conf, mconf, link_sta);
+		}
 		/* starec ht */
 		mt7996_mcu_sta_ht_tlv(skb, link_sta);
 		/* starec vht */
@@ -3274,8 +3278,6 @@ int mt7996_mcu_add_sta(struct mt7996_dev *dev, struct ieee80211_bss_conf *conf,
 		mt7996_mcu_sta_eht_tlv(skb, link_sta);
 		/* starec muru */
 		mt7996_mcu_sta_muru_tlv(dev, skb, conf, mconf, link_sta);
-		/* starec bfee */
-		mt7996_mcu_sta_bfee_tlv(dev, skb, conf, mconf, link_sta);
 
 		if (sta->mlo) {
 			/* starec mld setup */
-- 
2.45.2

